README.md

```diff
@@ -1,15 +1,106 @@
 # bidi-har-export
 
-Experimental module to compile WebDriver BiDi network events as a HAR file
+Experimental module to compile WebDriver BiDi network events as a HAR file. The WebDriver BiDi network events are currently only available in Firefox 110 or newer.
 
-## how to use
+## Examples
 
-```
+### HarRecorder
+
+`HarRecorder` is a base class for recording HAR. The consumer is responsible for indicating the beginning and end of the recording by using the `startRecording` and `stopRecording` methods. During the recording the consumer should forward all the relevant BiDi events to the recorder. The useful events for HAR generation are:
+- network.beforeRequestSent
+
+- network.responseCompleted
+- browsingContext.domContentLoaded
+- browsingContext.load
+
+The recorder will not handle subscribing or unsubscribing to those events. For selenium users, the `SeleniumBiDiHarRecorder` can be used to facilitate the process.
+
+```javascript
 const { HarRecorder } = require("bidi-har-export");
 
 const harRecorder = new HarRecorder({ browser: "firefox", version: "111.0a1"});
+
+// It is mandatory to call startRecording before forwarding any event to the
+// recorder.
 harRecorder.startRecording();
-// Forward network and browser context events to the recorder
-harRecorder.recordEvent(evt);
+
+// Forward network and browser context events to the recorder.
+// The recorder expects events as objects containing a "method"
+// and a "params" property.
+harRecorder.recordEvent(beforeRequestSentEvent);
+harRecorder.recordEvent(domContentLoadedEvent);
+harRecorder.recordEvent(loadEvent);
+harRecorder.recordEvent(responseCompletedEvent);
+
+// To complete the recording and generate the HAR record,
+// call stopRecording().
 const harRecord = harRecorder.stopRecording();
+
+// `harRecord` is an object following the HAR specification at
+// https://w3c.github.io/web-performance/specs/HAR/Overview.html
+// It is an object with a single property at the root called `log`
+// See the specification for more details.
+```
+
+### SeleniumBiDiHarRecorder
+
+The `SeleniumBiDiHarRecorder` can be used with a selenium node driver in order to facilitate a recording. This recorder will take care of subscribing and unsubscribing to the necessary event, so that the consumer only has to start and stop the recording.
```

Comment:
```suggestion
The `SeleniumBiDiHarRecorder` can be used with the [JavaScript binding for Selenium](https://www.selenium.dev/selenium/docs/api/javascript/) in order to facilitate a recording. This recorder will take care of subscribing and unsubscribing to the necessary event, so that the consumer only has to start and stop the recording.
```